### PR TITLE
Convert experimental primitive `TYPEREP_TYCON_NAME` into proper LF builtin

### PR DESCRIPTION
**Pull request by akrmn**
Closes https://github.com/digital-asset/daml/issues/13452

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/archive/src/main/protobuf/com/daml/daml_lf_dev/daml_lf_1.proto [hunk 1]
```diff
@@ -568,7 +568,9 @@ enum BuiltinFunction {
   NUMERIC_TO_BIGNUMERIC = 145; // *Available in versions >= 1.13*
   BIGNUMERIC_TO_TEXT = 146; // *Available in versions >= 1.13*
 
-  // Next id is 148. 147 is ANY_EXCEPTION_MESSAGE.
+  TYPEREP_TYCON_NAME = 148; // *Available in versions >= 1.dev*
+
+  // Next id is 149.
```
Comment (sofiafaro-da) [inline daml-lf/archive/src/main/protobuf/com/daml/daml_lf_dev/daml_lf_1.proto:573]:
```suggestion
  // Next id is 149. 148 is TYPEREP_TYCON_NAME.
```

**Review by sofiafaro-da — APPROVED** [review]
Looks great, thank you!

Could you also add a blurb for the new builtin in the LF spec, please? There's a section near the bottom where it goes over each of the builtins. This could be a separate PR.

File: daml-lf/archive/src/main/protobuf/com/daml/daml_lf_dev/daml_lf_1.proto (hunk 1 above)
Comment (akrmn) [reply daml-lf/archive/src/main/protobuf/com/daml/daml_lf_dev/daml_lf_1.proto]: Added it; I'd left it out since 147 was in a different part of the file

**Comment by akrmn** [comment]
> Could you also add a blurb for the new builtin in the LF spec

Will do, thanks for the reminder

**Review by remyhaemmerle-da — APPROVED** [review]

**Pull request merged**